airflow/executors/kubernetes_executor.py

```diff
@@ -572,9 +572,28 @@ def _labels_to_key(self, labels):
         )
         return None
 
+    def _flush_watcher_queue(self):
+        self.log.debug('Executor shutting down, watcher_queue approx. size=%d', self.watcher_queue.qsize())
+        while True:
+            try:
+                task = self.watcher_queue.get_nowait()
+                # Ignoring it since it can only have either FAILED or SUCCEEDED pods
+                self.log.warning('Executor shutting down, IGNORING watcher task=%s', task)
+                self.watcher_queue.task_done()
+            except Empty:
+                break
+
     def terminate(self):
         """Termninates the watcher."""
+        self.log.debug("Terminating kube_watcher...")
+        self.kube_watcher.terminate()
+        self.kube_watcher.join()
+        self.log.debug("kube_watcher=%s alive?=%s", self.kube_watcher, self.kube_watcher.is_alive())
```

Comment:
This isn't useful - after a call to `.join()` the kube_watcher should never be alive, right?

Reply:
Removed `is alive?` and rebased.